stdlib/public/core/HashedCollections.swift.gyb

```diff
@@ -575,19 +575,31 @@ public struct Set<Element : Hashable> :
   public mutating func intersectInPlace<
     S : SequenceType where S.Generator.Element == Element
   >(sequence: S) {
-    // Because `intersect` needs to both modify and iterate over
-    // the left-hand side, the index may become invalidated during
-    // traversal so an intermediate set must be created.
-    //
-    // FIXME(performance): perform this operation at a lower level
-    // to avoid invalidating the index and avoiding a copy.
-    let result = self.intersect(sequence)
-
-    // The result can only have fewer or the same number of elements.
-    // If no elements were removed, don't perform a reassignment
-    // as this may cause an unnecessary uniquing COW.
-    if result.count != count {
-      self = result
+    if case .Native(let nativeOwner) = _variantStorage {
+      // perform this operation at a lower level
+      // to avoid invalidating the index and avoiding a copy.
+      let native = nativeOwner.nativeStorage
+      var bucket = 0
+      while bucket < native.capacity {
+        if native.isInitializedEntry(bucket) &&
+            !sequence.contains(native.keyAt(bucket)) {
```

Comment:
`SequenceType.contains()` consumes the sequence.  If our tests didn't catch this, it means this patch needs to add tests that use `intersectInPlace()` with `MinimalSequence` as the argument.

To fix this issue, you need to use the `_preprocessingPass()` method and only query the sequence directly multiple times when it is multi-pass.  Otherwise, the sequence needs to be copied to a contiguous array first.